server/test/forms/translation/ProgramTranslationFormTest.java

```diff
@@ -0,0 +1,195 @@
+package forms.translation;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static play.test.Helpers.fakeRequest;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.Locale;
+import java.util.Optional;
+import models.Program;
+import org.junit.Test;
+import play.data.FormFactory;
+import play.mvc.Http.Request;
+import repository.ResetPostgres;
+import services.LocalizedStrings;
+import services.program.LocalizationUpdate;
+import services.program.StatusDefinitions;
+import support.ProgramBuilder;
+
+public class ProgramTranslationFormTest extends ResetPostgres {
+
+  private static final ImmutableMap<String, String> REQUEST_DATA_WITH_TWO_TRANSLATIONS =
+      ImmutableMap.<String, String>builder()
+          .put(ProgramTranslationForm.DISPLAY_NAME_FORM_NAME, "display name")
+          .put(ProgramTranslationForm.DISPLAY_DESCRIPTION_FORM_NAME, "display description")
+          .put(ProgramTranslationForm.configuredStatusFieldName(0), "first status configured text")
+          .put(ProgramTranslationForm.statusTextFieldName(0), "first status text")
+          .put(ProgramTranslationForm.statusEmailFieldName(0), "first status email")
+          .put(ProgramTranslationForm.configuredStatusFieldName(1), "second status configured text")
+          .put(ProgramTranslationForm.statusTextFieldName(1), "second status text")
+          .put(ProgramTranslationForm.statusEmailFieldName(1), "second status email")
+          .build();
+
+  @Test
+  public void bindFromRequest() throws Exception {
+    Request request = fakeRequest().bodyForm(REQUEST_DATA_WITH_TWO_TRANSLATIONS).build();
+
+    ProgramTranslationForm form =
+        ProgramTranslationForm.bindFromRequest(request, instanceOf(FormFactory.class), 2);
+    assertThat(form.getUpdateData())
+        .isEqualTo(
+            LocalizationUpdate.builder()
+                .setLocalizedDisplayName("display name")
+                .setLocalizedDisplayDescription("display description")
+                .setStatuses(
+                    ImmutableList.of(
+                        LocalizationUpdate.StatusUpdate.builder()
+                            .setConfiguredStatusText("first status configured text")
+                            .setLocalizedStatusText(Optional.of("first status text"))
+                            .setLocalizedEmailBody(Optional.of("first status email"))
+                            .build(),
+                        LocalizationUpdate.StatusUpdate.builder()
+                            .setConfiguredStatusText("second status configured text")
+                            .setLocalizedStatusText(Optional.of("second status text"))
+                            .setLocalizedEmailBody(Optional.of("second status email"))
```

Comment:
remove this line to evaluate the missing email scenario?

Reply:
I'm not sure I follow. In this case, the request data indicates that there's a configured email (by it's presence). The "empty-or-unrovided-email" scenario is already tested in `bindFromRequest_emptyStatusOrEmailIsConsideredNotProvided`